### PR TITLE
[KED-2860] Run query resolvers

### PR DESCRIPTION
**Pull request by merelcht**
## Description

https://jira.quantumblack.com/browse/KED-2860

## Development notes

* Added resolvers for fetching all runs and fetching a list of runs by IDs
* The schema is compliant with what we discussed yesterday, but I'll do a final check and type cleanup when this PR and https://github.com/quantumblacklabs/kedro-viz/pull/602 have been merged 
* This is built on top of the session store code so I've used that as a base for this PR 

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [X] Added tests to cover my changes

## Legal notice

- [X] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: package/kedro_viz/api/graphql.py [hunk 1]
```diff
@@ -42,111 +42,97 @@
...
     Returns:
         list of Run objects
-
     """
-    return [
-        RunModelGraphQLType(id=kedro_session.id, blob=kedro_session.blob)
-        for kedro_session in data_access_manager.db_session.query(RunModel).all()
-    ]
+    runs = []
+    session = data_access_manager.db_session
```
Comment (limdauto) [inline package/kedro_viz/api/graphql.py:91]: This could be None. Need to handle that.

File: package/kedro_viz/api/graphql.py [hunk 2]
```diff
@@ -42,111 +42,97 @@
...
-    """RunDetails object format"""
+class TrackingDataSet:
+    """TrackingDataSet object to structure tracking data for a Run."""
 
-    id: ID
-    name: str
-    details: str
+    datasetName: Optional[str]
+    datasetType: Optional[str]
+    data: Optional[str]
```
Comment (limdauto) [inline package/kedro_viz/api/graphql.py:119]: Why are all these optional?

**Review by limdauto — COMMENTED** [review]
LGTM mostly 🎉 Just a few minor tweaks needed

File: package/kedro_viz/api/graphql.py (hunk 2 above)
Comment (MerelTheisenQB) [reply package/kedro_viz/api/graphql.py:119]: This was still my doing, but matches the schema on the frontend:
```
type TrackingDataset {
    datasetName: String
    datasetType: String
    data: JSONObject
  }
```

File: package/kedro_viz/api/graphql.py [hunk 3]
```diff
@@ -41,112 +42,113 @@
...
+    runs_list: List[Run] = strawberry.field(resolver=get_all_runs)
 
-    runs: List[Run] = strawberry.field(resolver=get_runs)
-    all_runs: typing.List[RunModelGraphQLType] = strawberry.field(resolver=get_all_runs)
+    @strawberry.field
+    def run_metadata(self, run_ids: List[ID]) -> List[Run]:
+        """Query to get data for specific runs from the session store"""
+        runs = []
+        for run_id in run_ids:
+            run = get_run(run_id)
```
Comment (limdauto) [inline package/kedro_viz/api/graphql.py:148]: Okay this is a useful thing to remember: when we write ORM code, we need to be mindful of how many queries we are issuing again the database. In this case, every run_id will trigger a `get_run` which is 1 db query. It's much better to select multiple run ids with a single query from the db using [filter](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.filter) instead.

I'd personally change `get_run` into `get_runs(run_ids: List[ID])` for this purpose.

File: package/kedro_viz/data_access/managers.py [hunk 4]
```diff
@@ -81,11 +81,15 @@ def __init__(self):
     @property
     def db_session(self):  # pragma: no cover
         """Sqlite db connection session"""
+        if not self._db_session:
+            logger.warning(
+                    """Database connection was unsuccessful"""
```
Comment (limdauto) [inline package/kedro_viz/data_access/managers.py:86]: Formatting here is odd. No need for `"""`

File: package/kedro_viz/data_access/managers.py [hunk 5]
```diff
@@ -81,11 +81,15 @@ def __init__(self):
...
+                )
         return self._db_session
 
     @db_session.setter
     def db_session(self, db_session: DatabaseSession):
-        self._db_session = db_session
+            self._db_session = db_session
```
Comment (limdauto) [inline package/kedro_viz/data_access/managers.py:92]: This identation is off.

**Review by limdauto — APPROVED** [review]
LGTM!

**Review by rashidakanchwala — APPROVED** [review]

**Pull request merged**